rpd_generator/bdl_structure/bdl_commands/zone.py

```diff
@@ -320,6 +320,52 @@ def populate_data_elements(self):
                     zone_ef_power_per_flow * exhaust_airflow
                 )
             return
+        # Populate Terminal for terminals_has_demand_control_ventilation
+        if self.keyword_value_pairs.get(BDL_ZoneKeywords.OA_FLOW_PER) is not None:
+            if (
+                self.parent.keyword_value_pairs.get(BDL_SystemKeywords.DOA_SYSTEM)
+                is not None
+            ):
+                doas = self.parent.keyword_value_pairs.get(
+                    BDL_SystemKeywords.DOA_SYSTEM
+                )
+                if (
+                    self.parent.keyword_value_pairs.get(
+                        BDL_SystemKeywords.MIN_OA_METHOD
+                    )
+                    == "DCV-ZONE-SENSORS"
+                ):
+                    self.terminals_has_demand_control_ventilation = [False, False, True]
+                else:
+                    if (
+                        self.parent.keyword_value_pairs.get(
+                            BDL_SystemKeywords.DOAS_ATTACHED_TO
+                        )
+                        == "AHU-MIXED-AIR"
+                        and self.parent.keyword_value_pairs.get(
+                            BDL_SystemKeywords.MIN_OA_METHOD
+                        )
+                        == "DCV-RETURN-SENSORS"
+                    ):
+                        self.terminals_has_demand_control_ventilation = [
```

Comment:
are you sure you want to set the values for all 3 terminal types in this part of the logic? Or do you only care about setting `self.terminals_has_demand_control_ventilation[2] = True`?

Reply:
@JacksonJ-KC It is an OR. Either there is a DOAS or a System providing OA and DCV so I think we can set all of tthree,. Do you see a problem with this?

Reply:
Addressed